src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacFieldAndReferencedFieldMismatch/DacFieldAndReferencedFieldMismatchAnalyzer.cs

```diff
@@ -0,0 +1,220 @@
+					|| ContainsAttributeMatchingOneOf(x, pxContext.FieldAttributes.UnboundDataTypeAttributes));
+
+				// agreed to stop checking property here as the code may be incomplete at this point
+				if (foreignTypeAttribute is null)
+					return;
+
+				var localTypeAttribute = localTypeAttributes.FirstOrDefault(x => HaveDifferentFieldSize(x, foreignTypeAttribute, pxContext));
+
+				if (localTypeAttribute is not null)
+				{
+					ReportTypeSizeMismatch(
+						context, pxContext, localTypeAttribute, property.Name,
+						foreignField.Name, GetFieldSize(foreignTypeAttribute, pxContext));
+				}
+			}
+		}
+
+		private static bool IsAttributeIncompatibleWithCLRType(DacFieldAttributeInfo typeAttribute, ITypeSymbol clrType, PXContext pxContext)
+		{
+			return typeAttribute.FlattenedAcumaticaAttributes
+				.Select(x => pxContext.FieldAttributes.TypeAttributesToCLRTypesMap.GetValueOrDefault(x.Type))
+				.Where(x => x is not null)
+				.Any(x => !SymbolEqualityComparer.Default.Equals(x, clrType));
+		}
+
+		private DacSemanticModel? GetOrInferModel(INamedTypeSymbol type, PXContext pxContext, CancellationToken cancellation)
+		{
+			return _modelsCache.TryGetValue(type, out var model)
+				? model
+				: _modelsCache[type] = DacSemanticModel.InferModel(pxContext, type, cancellation: cancellation);
+		}
+
+		private static bool HaveDifferentUnderlyingType(DacPropertyInfo localProperty, ITypeSymbol clrType) =>
+			!SymbolEqualityComparer.Default.Equals(localProperty.PropertyType, clrType);
+
+		private static INamedTypeSymbol? ExtractForeignFieldSymbol(DacFieldAttributeInfo referenceAttribute, PXContext pxContext)
+		{
+			var argument = referenceAttribute.AttributeData.ConstructorArguments.FirstOrDefault();
+			INamedTypeSymbol? result;
+
+			if (argument.Value is INamedTypeSymbol { IsGenericType: true } symbol)
+				result = ExtractForeignFieldSymbolFromBqlSearch(symbol, pxContext);
+			else
+				result = argument.Value as INamedTypeSymbol;
+
+			return result?.ContainingSymbol is INamedTypeSymbol ? result : null;
+		}
+
+		private static INamedTypeSymbol? ExtractForeignFieldSymbolFromBqlSearch(INamedTypeSymbol searchType, PXContext pxContext)
+		{
+			return searchType.GetBaseTypesAndThis()
+							 .TakeWhile(x => x.SpecialType != SpecialType.System_Object)
+							 .OfType<INamedTypeSymbol>()
+							 .Reverse()
+							 .Where(x => x.IsGenericType && x.ImplementsInterface(pxContext.BQL.IBqlSearch))
+							 .Select(type => type.TypeArguments.FirstOrDefault(arg => arg.IsDacBqlField(pxContext)))
+							 .OfType<INamedTypeSymbol>()
+							 .FirstOrDefault();
+		}
+
+		private static int GetFieldSize(DacFieldAttributeInfo attrInfo, PXContext pxContext)
+		{
+			if (ContainsAttributeMatchingOneOf(attrInfo, pxContext.FieldAttributes.DataAttributesWithLength)
+				&& attrInfo.AttributeData.ConstructorArguments.FirstOrDefault().Value is int i)
+				return i;
+
+			var lengthCtorArguments = attrInfo.AttributeType
```

Comment:
@SENya1990, using `FlattenedAcumaticaAttributes` with custom aggregated attribute didn't reveal PXString/PXDBString that will be applied as far as I understand - thus the usage of the function. And no, the point is not in ctor analysis, I thought we already agreed on this one.

Reply:
@acasj `FlattenedAcumaticaAttributes` should contain all attributes applied to the DAC field. They are recursively collected from the attribute that is actually declared on the field. This includes:
- base attribute types
- aggregated attributes
- base types of aggregated attributes
- attributes aggregated on aggregated attributes, their base types and so on.

If an attribute is derived from a `PXString`/`PXDBString` attribute, or the attribute aggregates it, then `FlattenedAcumaticaAttributes` should already have it.  

If you have a concrete example of a case when the attribute is derived from `PXString`/`PXDBString`, or aggregates it somehow, please provide it here. This would mean that there is a bug in the collection of the `FlattenedAcumaticaAttributes` set which should be fixed. I will create a bug for such case.

Reply:
@SENya1990, 
1. The example is in the tests: `PropertiesWithWrongLengthAreMarkedAsErroneous`, field `Aggregated` have only `PXDBString` in `FlattenedAcumaticaAttribute`
2. Moreover, I don't see constructor arguments there for `PXDBString` attribute - `Application.*Arguments` are all empty

Reply:
I will take a look and investigate this issue. If there is a bug, I think I will make a separate bug in Acuminator.

Reply:
@acasj did you mean these two attributes:
![image](https://github.com/user-attachments/assets/e230ad1d-be93-4eb9-8ecf-8d5543fc9297)
These attributes are not aggregator attributes, so this declaration is incorrect because not all Acumatica attributes support aggregation of attributes.

Aggregator attributes (or aggregators, in short) are only those attributes that are derived from `PX.Data.PXAggregateAttribute`.

My fault, I missed it during reviews because I looked more closely at the analyzer code. I will add a separate remark for this test source.

Reply:
This remark wasn't fully addressed, I will describe more in additional remarks